tests/model/test_model.py

```diff
@@ -2048,6 +2048,99 @@ def test_notify_users_hides_PM_content_based_on_user_setting(
             f"Test Organization Name:\nFoo Foo (to you{others})", expected_content
         )
 
+    @pytest.mark.parametrize(
+        [
+            "user_id",
+            "flags",
+            "visual_notification_status",
+            "display_recipient",
+            "subject",
+            "is_stream_muted",
+            "is_topic_muted",
+            "no_of_times_notify_call_expected",
+        ],
+        [
+            (
+                4444,
+                {"flags": ["has_alert_word"]},
+                True,
+                {"display_recipient": ["test_here"]},
+                {"subject": ["k1"]},
+                True,
+                True,
+                1,
+            ),
+            (
+                4444,
+                {"flags": ["has_alert_word"]},
+                False,
+                {"display_recipient": ["test_here"]},
+                {"subject": ["k1"]},
+                True,
+                True,
+                0,
+            ),
+            (
+                4444,
+                {"flags": ["has_alert_word"]},
+                False,
+                {"display_recipient": ["test_here"]},
+                {"subject": ["k1"]},
+                True,
+                False,
+                0,
+            ),
+            (
+                4444,
+                {"flags": ["has_alert_word"]},
+                False,
+                {"display_recipient": ["test_here"]},
+                {"subject": ["k1"]},
+                False,
+                False,
+                1,
+            ),
+        ],
+        ids=[
+            "visual_notification is on -> no effect due to mute",
+            "visual_notification is off, and both topic and stream is muted",
+            "visual_notification is off, only topic is muted",
+            "visual_notification is off, neither topic or stream is muted",
+        ],
+    )
+    def test_notify_user_with_alert_words(
+        self,
+        mocker,
+        model,
+        stream_msg_fixture,
+        user_id,
+        flags,
+        visual_notification_status,
+        display_recipient,
+        subject,
+        is_stream_muted,
+        is_topic_muted,
+        no_of_times_notify_call_expected,
+    ):
+        stream_msg_fixture.update(flags)
+        self.controller.notify_enabled = True
+        stream_msg_fixture.update(display_recipient)
+        stream_msg_fixture.update(subject)
+        mocker.patch.object(
+            model,
+            "is_visual_notifications_enabled",
+            return_value=visual_notification_status,
+        )
+        mocker.patch.object(model, "is_muted_stream", return_value=is_stream_muted)
+        mocker.patch.object(model, "is_muted_topic", return_value=is_topic_muted)
+        notify = mocker.patch(MODULE + ".notify")
+        model.notify_user(stream_msg_fixture)
+        if no_of_times_notify_call_expected == 1:
+            assert notify.called
```

Comment:
Maybe, the parameters with which the `notify` method would also benefit from testing? If you decide to do that, you can use `notify.assert_called_once_with(param1,..)` and `notify.assert_not_called()`.

Reply:
Yep! agreed.